python_bindings/src/builtin_helpers_src.py

```diff
@@ -0,0 +1,787 @@
+            setattr(self._gp, name, new_value)
+        else:
+            self._unhandled_generator_params[name] = value
+
+    def _build_pipeline(self) -> Pipeline:
+        if self._stage <= _Stage.IOFinalized:
+            self._finalize_io()
+
+        assert self._stage == _Stage.IOFinalized
+        assert not self._pipeline
+        assert not self._input_parameters
+        assert not self._output_funcs
+
+        self.generate(self._gp, self._inputs, self._outputs)
+        self.schedule(self._gp, self._inputs, self._outputs)
+
+        self._input_parameters = {
+            n: getattr(self._inputs, n).parameter()
+            for n in self._inputs_names
+        }
+        self._output_funcs = {
+            n: getattr(self._outputs, n)
+            for n in self._outputs_names
+        }
+        _check(
+            len(self._output_funcs) > 0,
+            "Generator %s must declare at least one Output in Arguments." %
+            self._get_name(),
+        )
+
+        funcs = []
+        for name, f in self._output_funcs.items():
+            _check(f.defined(), "Output '%s' was not defined." % name)
+            self._requirements[name]._check_types_and_dims(
+                f.output_types(), f.dimensions())
+            funcs.append(f)
+
+        self._pipeline = Pipeline(funcs)
+        self._stage = _Stage.PipelineBuilt
+        return self._pipeline
+
+    def _get_input_parameter(self, name: str) -> list[InternalParameter]:
+        assert self._stage == _Stage.PipelineBuilt
+        _check(name in self._input_parameters, "Unknown input: %s" % name)
+        return [self._input_parameters[name]]
+
+    def _get_output_func(self, name: str) -> list[Func]:
+        assert self._stage == _Stage.PipelineBuilt
+        _check(name in self._output_funcs, "Unknown output: %s" % name)
+        return [self._output_funcs[name]]
+
+    def _bind_input(self, name: str, values: list[object]):
+        assert (self._stage == _Stage.GeneratorParamsBuilt
+                or self._stage == _Stage.IOBuilt)
+        _check(
+            len(values) == 1, "Too many values specified for input: %s" % name)
+        _check(name in self._inputs_names,
+               "There is no input with the name: %s" % name)
+        g = getattr(self._inputs, name)
+        assert not name in self._replacements
+        self._replacements[name] = g._make_replacement(
+            values[0], self._requirements[name])
+
+
+_python_generators: dict = {}
+
+
+def _get_python_generator_names() -> list[str]:
+    return _python_generators.keys()
+
+
+def _fqname(o):
+    k = o
+    m = k.__module__
+    q = k.__qualname__
+    if m == "__main__" or k == "builtins":
+        return q
+    return m + "." + q
+
+
+def _find_python_generator_class(name: str):
+    cls = _python_generators.get(name, None)
+    if not isclass(cls):
+        cls = None
+    return cls
+
+
+def alias(**kwargs):
+
+    def alias_impl(cls):
+        nonlocal kwargs
```

Comment:
This isn't necessary since you aren't writing to `kwargs`. It's already captured read-only, enough for calling `.items()`.